openshift/4.0/templates/monitor/deploy-grafana.yaml

```diff
@@ -0,0 +1,1298 @@
+                    "text": "1d",
+                    "value": "1d"
+                  },
+                  {
+                    "selected": false,
+                    "text": "7d",
+                    "value": "7d"
+                  },
+                  {
+                    "selected": false,
+                    "text": "14d",
+                    "value": "14d"
+                  },
+                  {
+                    "selected": false,
+                    "text": "30d",
+                    "value": "30d"
+                  }
+                ],
+                "query": "1m,5m,10m,30m,1h,6h,12h,1d,7d,14d,30d",
+                "refresh": 2,
+                "skipUrlSync": false,
+                "type": "interval"
+              },
+              {
+                "allValue": null,
+                "current": {
+                  "selected": false,
+                  "text": "All",
+                  "value": "$__all"
+                },
+                "datasource": "Prometheus",
+                "definition": "label_values(http_requests_received_total, instance)",
+                "hide": 0,
+                "includeAll": true,
+                "label": "instances",
+                "multi": true,
+                "name": "instances",
+                "options": [],
+                "query": "label_values(http_requests_received_total, instance)",
+                "refresh": 1,
+                "regex": "",
+                "skipUrlSync": false,
+                "sort": 1,
+                "tagValuesQuery": "",
+                "tags": [],
+                "tagsQuery": "",
+                "type": "query",
+                "useTags": false
+              }
+            ]
+          },
+          "time": {
+            "from": "now-6h",
+            "to": "now"
+          },
+          "timepicker": {
+            "refresh_intervals": [
+              "5s",
+              "10s",
+              "30s",
+              "1m",
+              "5m",
+              "15m",
+              "30m",
+              "1h",
+              "2h",
+              "1d"
+            ],
+            "time_options": [
+              "5m",
+              "15m",
+              "1h",
+              "6h",
+              "12h",
+              "24h",
+              "2d",
+              "7d",
+              "30d"
+            ]
+          },
+          "timezone": "",
+          "title": "ALL",
+          "uid": "zyAf4i4Zz",
+          "version": 2
+        }
+
+  # Service account grafana will run under.
+  - kind: ServiceAccount
+    apiVersion: v1
+    metadata:
+      name: ${APP_NAME}
+      namespace: ${PROJECT_NAMESPACE}-${ENV_NAME}
+      labels:
+        name: ${APP_NAME}
+        app: ${APP_NAME}
+        role: ${ROLE_NAME}
+        env: ${ENV_NAME}
+      annotations:
+        serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"${APP_NAME}"}}'
```

Comment:
oauth does not currently work.  Need to find out if @goemen had to do additional steps previously to get it working, or whether the Lab still supports it going forward.